### ISSUE TITLE
Feasibility of this concept for armc6

### ISSUE THREAD

**Issue by 40Grit** [bogdanm/udynlink#2]
What are the performance considerations for what this project was/is trying to acheive?

Can it be ported to armclang?

Feasibility of building Mbed OS position independently?

**Comment by 0Grit** [comment]
@bogdanm

**Comment by bogdanm** [comment]
- No idea on the performance, I have not measured it. The whole thing is very much WIP and still in research phase, although some things are already working. The original use case for this project was to allow eLua to load binary modules at runtime, not only Lua modules.
- No idea, sorry.
- Not sure what you mean by "position independently". You can build an executable in PIC/PID mode independently of a dynamic linker, so not sure what use case you have in mind.

**Comment by 40Grit** [comment]
My thinking is runtime firmware updates for mission critical systems. 

Ex. Update vulnerable network stack without a system reset.

**Comment by 40Grit** [comment]
I was thinking this was adding an extra layer on top of pic/pid. I'll reread the readme and check out sources.

This is focusing more on dynamic linking before writing to flash?

**Comment by bogdanm** [comment]
No, this is focusing on both parts: creating a loadable module (that you can write to flash or anywhere else) and then loading and using it. Runtime firmware updates are definitely possible.
